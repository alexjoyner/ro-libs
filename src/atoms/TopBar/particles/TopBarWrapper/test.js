import React from 'react';
import { shallow } from 'enzyme';
import { TopBarWrapper } from './';

describe('TopBarWrapper component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<TopBarWrapper />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should display apply correct props when not sticky', () => {
    expect(component).not.toHaveStyleRule('z-index', '200');
    expect(component).not.toHaveStyleRule('position', 'fixed');
    expect(component).not.toHaveStyleRule('top', '0');
  });
  it('Should display apply correct props when sticky', () => {
    const testSticky = shallow(<TopBarWrapper sticky />);
    expect(testSticky).toHaveStyleRule('z-index', '200');
    expect(testSticky).toHaveStyleRule('position', 'fixed');
    expect(testSticky).toHaveStyleRule('top', '0');
  });
  it('should default to a height of 67px', () => {
    expect(component).toHaveStyleRule('height', '67px');
  });
  it('should accept a height prop', () => {
    const testHeight = shallow(<TopBarWrapper height="100px" />);
    expect(testHeight).toHaveStyleRule('height', '100px');
  });
  it('should apply line height from height prop', () => {
    const testHeight = shallow(<TopBarWrapper height="100px" />);
    expect(testHeight).toHaveStyleRule('line-height', '100px');
  });
  it('should accept a color prop', () => {
    const testHeight = shallow(<TopBarWrapper color="blue" />);
    expect(testHeight).toHaveStyleRule('background-color', 'blue');
  });
});
