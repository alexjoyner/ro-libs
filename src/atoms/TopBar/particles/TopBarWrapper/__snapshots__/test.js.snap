// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarWrapper component Should match snapshot 1`] = `
.c0 {
  text-align: center;
  height: 67px;
  line-height: 67px;
  width: 100%;
}

<div
  className="c0"
/>
`;
